DrawMat/ViewModels/InteractionMode.cs

```diff
@@ -36,11 +37,15 @@ public void PointerReleased(MainViewModel vm, Point position)
     {
         if (_currentShape != null)
         {
-            vm.RootGroup.Children.Add(_currentShape);
+            vm.RootGroup.Add(_currentShape);
         }
         _currentShape = null;
     }
 
+    public void PointerPressedRight(MainViewModel vm, Point position)
+    {
+    }
```

Comment:
Maybe there should be a default implementation for all these methods, so that you only have to explicitly implement handlers for events you care about in a given mode.

Reply:
"all these methods" includes methods other than the one I commented on.